cmd/soroban-cli/src/commands/keys/generate.rs

```diff
@@ -49,6 +50,47 @@ pub struct Cmd {
 }
 
 impl Cmd {
+    async fn fund_identity(&self, secret: &Secret) -> Result<(), Error> {
+        if !self.no_fund {
+            let addr = secret.public_key(self.hd_path)?;
+            let network = self.network.get(&self.config_locator)?;
+            let network_name = get_network_name(&network.rpc_url).unwrap_or(&network.rpc_url);
+            eprintln!("🌎 Funding account with public key as address on {network_name}...",);
+            match network.fund_address(&addr).await {
+                Ok(()) => {
+                    eprintln!("✅ Funded (use 'stellar keys fund me' to fund again)");
+                }
+                Err(e) => {
+                    tracing::warn!("fund_address failed: {e}");
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn write_and_fund_identity(&self, secret: &Secret) -> Result<(), Error> {
+        self.config_locator.write_identity(&self.name, secret)?;
+        eprintln!("✅ Generated new key for '{}'", self.name);
+        eprintln!("ℹ️ Public key: {}", secret.public_key(self.hd_path)?);
+        eprintln!(
+            "ℹ️ Secret key: hidden (use 'stellar keys show {}' to view)",
+            self.name
+        );
+        self.fund_identity(secret).await?;
+        Ok(())
+    }
+
+    async fn handle_existing_identity(&self, existing_secret: &Secret) -> Result<(), Error> {
+        eprintln!("The identity {} already exists!", self.name);
+        if let Ok(root) = self.config_locator.config_dir() {
+            let mut path = root.join("identity").join(&self.name);
+            path.set_extension("toml");
+            eprintln!("    Seed phrase found at: {}", path.display());
+        }
+        self.fund_identity(existing_secret).await?;
```

Comment:
I don't think the command should fund an existing account. Unclear the purpose this serves since the generate command is primarily about getting a new key. If someone needs to top up their account they can use the fund command.

There's an open issue to remove funding from the generate command:
- https://github.com/stellar/stellar-cli/issues/1407

Reply:
I removed --no-fund and the funding portions.